Sources/NIO/Bootstrap.swift

```diff
@@ -966,4 +1135,141 @@ public final class NIOPipeBootstrap {
             }
         }
     }
+    
+    /// A channel option which can be applied to datagram bootstrap using shorthand notation.
+    /// - See: NIOPipeBootstrap.channelOptions(_ options: [Option])
+    public struct Option {
+        private let data: ShorthandChannelOption
+        
+        private init(_ data: ShorthandChannelOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied NIOPipeBootstrap
+        /// - Parameter to: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        @usableFromInline
+        func applyOption(to bootstrap: NIOPipeBootstrap) {
+            data.applyOption(to: bootstrap)
+        }
+        
+        fileprivate enum ShorthandChannelOption {
+            case disableAutoRead
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to bootstrap: NIOPipeBootstrap) {
+                switch self {
+                case .disableAutoRead:
+                    _ = bootstrap.channelOption(ChannelOptions.autoRead, value: false)
+                case .allowRemoteHalfClosure(let value):
+                    _ = bootstrap.channelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension NIOPipeBootstrap.Option: Hashable {}
+extension NIOPipeBootstrap.Option.ShorthandChannelOption: Hashable {}
+
+/// Approved shorthand datagram channel options.
+extension NIOPipeBootstrap.Option {
+    /// Option to disable autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = NIOPipeBootstrap.Option(.disableAutoRead)
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static let allowRemoteHalfClosure =
+        NIOPipeBootstrap.Option(.allowRemoteHalfClosure(true))
```

Comment:
This doesn't seem necessary given the function version of this below?

Reply:
No, but saves typing for the user and these are the lazy man's options.